### PR TITLE
feat(web): extend unit-test oriented dummy model 📚 

### PR DESCRIPTION
**Pull request by jahorton**
The old `DummyModel` used to mock models for automated testing was lacking support for a number of model features we'd like to unit test.  Since we're looking to unit-test the recently-refactored predictive-text component functions, it would be helpful to use it for mocking instead of requiring a fully-compiled model.  So, I thought it wise to give the old test-oriented class an update.  

@keymanapp-test-bot skip

### CODE REVIEW

File: common/web/lm-worker/src/main/models/dummy-model.ts [hunk 1]
```diff
@@ -30,15 +30,33 @@
...
  */
-export default class DummyModel implements LexicalModel {
+export class DummyModel implements LexicalModel {
   configuration: Configuration;
   punctuation?: LexicalModelPunctuation;
-  private _futureSuggestions: Suggestion[][];
+  toKey?: Wordform2Key;
+  applyCasing?: (form: CasingForm, text: string) => string;
+  wordbreaker?: WordBreakingFunction;
+  languageUsesCasing?: boolean;
```
Comment (jahorton) [inline common/web/lm-worker/src/main/models/dummy-model.ts:55]: The dummy model doesn't actually use these internally, of course.  The component methods for the prediction process, however, do use them - and so this makes a great spot to mock the requirements for those methods.

File: common/models/templates/src/common.ts [hunk 2]
```diff
@@ -115,10 +115,13 @@ export function transformToSuggestion(transform: Transform, p: number): WithOutc
 export function transformToSuggestion(transform: Transform, p?: number): Outcome<Suggestion> {
   let suggestion: Outcome<Suggestion> = {
     transform: transform,
-    transformId: transform.id,
     displayAs: transform.insert
   };
 
+  if(transform.id !== undefined) {
+    suggestion.transformId = transform.id;
+  }
```
Comment (mcdurdin) [inline common/models/templates/src/common.ts:123]: Curious why this change is needed?

File: common/web/lm-worker/src/main/models/dummy-model.ts [hunk 3]
```diff
@@ -48,6 +66,22 @@ export default class DummyModel implements LexicalModel {
...
+      this.wordbreaker = options.wordbreaker;
+    }
+
+    if (options.applyCasing) {
+      this.applyCasing = options.applyCasing;
+    }
+
+    if (options.languageUsesCasing) {
+      this.languageUsesCasing = options.languageUsesCasing;
+    }
```
Comment (mcdurdin) [inline common/web/lm-worker/src/main/models/dummy-model.ts:84]: Any reason why we can't just shorten this?

```suggestion
    this.toKey = options.toKey;
    this.wordbreaker = options.wordbreaker;
    this.applyCasing = options.applyCasing;
    this.languageUsesCasing = options.languageUsesCasing;
```

The only reason I can think of for not doing this -- are we enumerating over properties in such a way that this would fail? If so, that seems like a second problem waiting!

If you are concerned about nullish tests, then you could:

```suggestion
    this.toKey = options.toKey ?? undefined;
    this.wordbreaker = options.wordbreaker ?? undefined;
    this.applyCasing = options.applyCasing ?? undefined;
    this.languageUsesCasing = options.languageUsesCasing ?? undefined;
```

**Review by mcdurdin — APPROVED** [review]

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 18.0.75-alpha](https://keyman.com/downloads/releases/alpha/18.0.75)
